packages/language/src/language-server/types.ts

```diff
@@ -138,28 +221,31 @@ export function diagnosticsToLSP(
   return map;
 }
 
-export function diagnosticToLSP(diagnostic: Diagnostic): lsp.Diagnostic {
+export function diagnosticToLSP(
+  diagnostic: Diagnostic,
+): lsp.Diagnostic | undefined {
+  if (
+    !diagnostic.uri ||
+    !diagnostic.range ||
+    isNaN(diagnostic.range.start) ||
+    isNaN(diagnostic.range.end)
+  ) {
+    return undefined;
+  }
   const doc = TextDocuments.get(diagnostic.uri);
+  if (!doc) {
+    return undefined;
+  }
   return {
     severity: severityToLsp(diagnostic.severity),
     range: {
-      start: doc
-        ? offsetToPosition(doc, diagnostic.range.start)
-        : {
-            character: 0,
-            line: 0,
-          },
-      end: doc
-        ? offsetToPosition(doc, diagnostic.range.end)
-        : {
-            character: 0,
-            line: 0,
-          },
+      start: offsetToPosition(doc, diagnostic.range.start),
+      end: offsetToPosition(doc, diagnostic.range.end),
     },
     message: diagnostic.message,
     code: diagnostic.code,
     data: diagnostic.data,
-    source: diagnostic.source ?? "pli",
+    source: diagnostic.source ?? "PL/I",
   };
```

Comment:
I was wondering here, what task this string has to fulfill? Is it meant as a language ID? Where is it used or displayed? In the context of PL/I we also have "lexer" as source I think. So, "PL/I" or "pli" are just fallback string to be displayed later? Or do they need to point to something like an ID would do.

Reply:
The `source` is shown in the UI and presented to the user:

<img width="724" height="93" alt="image" src="https://github.com/user-attachments/assets/846a8541-23ed-4bfa-b17f-189c96777cc6" />

I've removed `lexer` from the `source` a while ago, because the `source` is supposed to be used to inform the user about which extension is sourcing the diagnostic. We use the `code` for more specific information.